tfhe/src/typed_api/booleans/types/static_.rs

```diff
@@ -58,6 +58,15 @@ pub(in crate::typed_api::booleans) type FheBoolServerKey =
 pub(in crate::typed_api::booleans) type FheBoolPublicKey =
     GenericBoolPublicKey<StaticBoolParameters>;
 
+#[derive(Clone, Debug, ::serde::Deserialize, ::serde::Serialize)]
+pub(in crate::typed_api::booleans) struct FheBoolCompressedPublicKey;
+
+impl FheBoolCompressedPublicKey {
+    pub(crate) fn new(_client_key: &FheBoolClientKey) -> Self {
+        Self
+    }
+}
```

Comment:
Its because the booleans does not have compressed public key, to I use a zero sized struct instead. Its the quickest way to work around the fact that boolean do not have a compressed public key type

The `CompressedPublicKey` type that regroups all compressed public keys must have one key for each type, and these keys must have a `new` method.

In the end, the user will have a compile time error it tries to encrypt a FheBool using a CompressedPublicKey due to the 
`FheBoolCompressedPublicKey` not implementing the needed traits.

Reply:
ah damn, don't know how I missed the compressed pk for boolean at the time, I'll create an issue for it

Reply:
https://github.com/zama-ai/tfhe-rs-internal/issues/83